chainer/optimizer.py

```diff
@@ -518,3 +523,50 @@ def __call__(self, opt):
                 grad = param.grad
                 with cuda.get_device(grad):
                     grad *= rate
+
+
+class GradientNoise(object):
+    """Optimizer hook function for adding gradient noise.
+
+    This hook function simply adds time-dependent annealed Gaussian noise
+    to the gradient at every training step:
+
+    . math::
+
+        g_t \\leftarrow g_t + N(0, \\sigma_t^2)
```

Comment:
As we design this hook so that we can flexibly change the noise function (by `noise_func`), we need to change the document so that this behavior is the default one. We also need to write how to change the noise function as well.